livekit-agents/livekit/agents/beta/workflows/warm_transfer.py

```diff
@@ -0,0 +1,352 @@
+from ...log import logger
+from ...types import NOT_GIVEN, NotGivenOr
+from ...utils import is_given
+from ...voice import room_io
+from ...voice.agent import Agent, AgentTask
+from ...voice.agent_session import AgentSession
+from ...voice.background_audio import (
+    AudioConfig,
+    AudioSource,
+    BackgroundAudioPlayer,
+    BuiltinAudioClip,
+    PlayHandle,
+)
+
+if TYPE_CHECKING:
+    from ...voice.audio_recognition import TurnDetectionMode
+
+
+BASE_INSTRUCTIONS = """
+# Identity
+
+You are an agent that is reaching out to a human agent for help. There has been a previous conversation
+between you and a caller, the conversation history is included below.
+
+# Goal
+
+Your main goal is to give the human agent sufficient context about why the caller had called in,
+so that the human agent could gain sufficient knowledge to help the caller directly.
+
+# Context
+
+In the conversation, user refers to the human agent, caller refers to the person who's transcript is included.
+Remember, you are not speaking to the caller right now, you are speaking to the human agent.
+
+Once the human agent has confirmed, you should call the tool `connect_to_caller` to connect them to the caller.
+
+Start by giving them a summary of the conversation so far, and answer any questions they might have.
+
+## Conversation history with caller
+{conversation_history}
+## End of conversation history with caller
+
+You are talking to the human agent now,
+give a brief introduction of the conversation so far, and ask if they want to connect to the caller.
+"""
+
+
+@dataclass
+class WarmTransferResult:
+    human_agent_identity: str
+
+
+class WarmTransferTask(AgentTask[WarmTransferResult]):
+    def __init__(
+        self,
+        target_phone_number: str,
+        *,
+        hold_audio: NotGivenOr[AudioSource | AudioConfig | list[AudioConfig] | None] = NOT_GIVEN,
+        sip_trunk_id: NotGivenOr[str] = NOT_GIVEN,
+        extra_instructions: str = "",
+        chat_ctx: NotGivenOr[llm.ChatContext] = NOT_GIVEN,
+        turn_detection: NotGivenOr[TurnDetectionMode | None] = NOT_GIVEN,
+        tools: NotGivenOr[list[llm.FunctionTool | llm.RawFunctionTool]] = NOT_GIVEN,
+        stt: NotGivenOr[stt.STT | None] = NOT_GIVEN,
+        vad: NotGivenOr[vad.VAD | None] = NOT_GIVEN,
+        llm: NotGivenOr[llm.LLM | llm.RealtimeModel | None] = NOT_GIVEN,
+        tts: NotGivenOr[tts.TTS | None] = NOT_GIVEN,
+        allow_interruptions: NotGivenOr[bool] = NOT_GIVEN,
+    ) -> None:
+        super().__init__(
+            instructions=self.get_instructions(
+                chat_ctx=chat_ctx, extra_instructions=extra_instructions
+            ),
+            chat_ctx=NOT_GIVEN,  # don't pass the chat_ctx
+            turn_detection=turn_detection,
+            tools=tools or [],
+            stt=stt,
+            vad=vad,
+            llm=llm,
+            tts=tts,
+            allow_interruptions=allow_interruptions,
+        )
+
```

Comment:
Can we add something in the constructor checking for the JobContext and raising an error if this isn't using Cloud (MoveParticipant is only supported on Cloud)